ttnn/cpp/ttnn/operations/experimental/ccl/ccl_op_fusion.hpp

```diff
@@ -0,0 +1,117 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "tt_metal/host_api.hpp"
+#include "tt_metal/impl/program/program.hpp"
+
+namespace ttnn {
+namespace experimental {
+namespace ccl {
+
+struct AllGatherFusedOpSignaler {
+
+    std::vector<CoreCoord> fused_op_receiver_cores;
+    std::vector<CoreCoord> fused_op_receiver_cores_noc;
+    std::vector<uint32_t> fused_op_receiver_signal_semaphores;
+
+    /* All Gather specific */
+    std::vector<CoreCoord> all_gather_worker_cores_noc;
+    uint32_t all_gather_worker_sync_semaphore;
+
+    bool initialized_fused_op = false;
+    bool initialized_all_gather = false;
+
+
+    AllGatherFusedOpSignaler(
+        std::vector<CoreCoord> fused_op_receiver_cores,
```

Comment:
pass by const reference.
Another way use can always use std::move inside of the constructor.
But I prefere 2 constructors, one is move and another with const reference.